Main.py

```diff
@@ -167,10 +169,12 @@ def main(args, seed=None, baked_server_options: Optional[Dict[str, object]] = No
 
     # remove starting inventory from pool items.
     # Because some worlds don't actually create items during create_items this has to be as late as possible.
-    if any(world.start_inventory_from_pool[player].value for player in world.player_ids):
+    if any(getattr(world.worlds[player].options, "start_inventory_from_pool", None) for player in world.player_ids):
         new_items: List[Item] = []
         depletion_pool: Dict[int, Dict[str, int]] = {
-            player: world.start_inventory_from_pool[player].value.copy() for player in world.player_ids}
+            player: getattr(world.worlds[player], "start_inventory_from_pool", StartInventoryPool({})).value.copy()
```

Comment:
this is missing the `.options`, so it'll always return empty. I think going forward we should maybe add optional core options to the dataclass somehow, but that's out of scope here.
```suggestion
            player: getattr(world.worlds[player].options,
                            "start_inventory_from_pool",
                            StartInventoryPool({})).value.copy()
```